libs/ui/tabs/brain/src/lib/brn-tabs-paginated-list.directive.ts

```diff
@@ -0,0 +1,662 @@
+{
+	abstract _items: QueryList<MatPaginatedTabHeaderItem>;
+	abstract _tabListContainer: ElementRef<HTMLElement>;
+	abstract _tabList: ElementRef<HTMLElement>;
+	abstract _tabListInner: ElementRef<HTMLElement>;
+	abstract _nextPaginator: ElementRef<HTMLElement>;
+	abstract _previousPaginator: ElementRef<HTMLElement>;
+
+	/** The distance in pixels that the tab labels should be translated to the left. */
+	private _scrollDistance = 0;
+
+	/** Whether the header should scroll to the selected index after the view has been checked. */
+	private _selectedIndexChanged = false;
+
+	private _root = inject(BrnTabsDirective);
+	private readonly _value = this._root.$value;
+	private readonly _tabs = this._root.$tabs;
+
+	/** Emits when the component is destroyed. */
+	protected readonly _destroyed = new Subject<void>();
+
+	/** Whether the controls for pagination should be displayed */
+	_showPaginationControls = false;
+
+	/** Whether the tab list can be scrolled more towards the end of the tab label list. */
+	_disableScrollAfter = true;
+
+	/** Whether the tab list can be scrolled more towards the beginning of the tab label list. */
+	_disableScrollBefore = true;
+
+	/**
+	 * The number of tab labels that are displayed on the header. When this changes, the header
+	 * should re-evaluate the scroll position.
+	 */
+	private _tabLabelCount!: number;
+
+	/** Whether the scroll distance has changed and should be applied after the view is checked. */
+	private _scrollDistanceChanged!: boolean;
+
+	/** Used to manage focus between the tabs. */
+	private _keyManager!: FocusKeyManager<MatPaginatedTabHeaderItem>;
+
+	/** Cached text content of the header. */
+	private _currentTextContent!: string;
+
+	/** Stream that will stop the automated scrolling. */
+	private _stopScrolling = new Subject<void>();
+
+	/**
+	 * Whether pagination should be disabled. This can be used to avoid unnecessary
+	 * layout recalculations if it's known that pagination won't be required.
+	 */
+	@Input({ transform: booleanAttribute })
+	disablePagination: boolean = false;
+
+	/** The index of the active tab. */
+	// @Input({ transform: numberAttribute })
+	// get selectedIndex(): number {
+	//   return this._selectedIndex;
+	// }
+	// set selectedIndex(v: number) {
+	//   const value = isNaN(v) ? 0 : v;
+
+	//   console.log("selectedIndex", value);
+
+	//   if (this._selectedIndex != value) {
+	//     this._selectedIndexChanged = true;
+	//     this._selectedIndex = value;
+
+	//     if (this._keyManager) {
+	//       this._keyManager.updateActiveItem(value);
+	//     }
+	//   }
+	// }
+	private _selectedIndex = computed(() => {
+		const currentKey = this._value();
+
+		let activeIndex = 0;
+		if (currentKey && this._items) {
+			// TODO currentTab is undefined for initial load
+			const currentTab = this._tabs[currentKey];
+			if (currentTab) {
+				activeIndex = this._items.toArray().indexOf(currentTab.trigger);
+			}
+		}
+
+		return activeIndex;
+	});
```

Comment:
I am calculating the `selectedIndex` based on the roots tab value. Then I need to get the current tab for the tab value to calculate the position of the tab. However, the `currentTab` is always `undefined` on initial load. Probably the tab value is available before the `tabs` have been registered.

Reply:
I think tabs was created pre signals. What if we make `tabs` a signal?

Reply:
Yes that did the trick! The initial tab is now correctly calculated and scrolled into view.